docs(css): Fx116 - Update "offset-position" property syntax

dipikabh opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description

The syntax of the [`offset-position`](https://developer.mozilla.org/en-US/docs/Web/CSS/offset-position) property has changed.
- A new keyword (`normal`) has been added: https://drafts.fxtf.org/motion/#offset-position-property

This PR updates the `offset-position` property page:
- Adds the new `normal` keyword
- Updates the description of the `auto` keyword
- Adds a "Result" frame for the existing example
- Adds a new example to show the difference in the values of the `offset-position` property (codepen: https://codepen.io/dipikabh/pen/oNQZyOm)

### Caveats

- Firefox has newly added support for the `offset-position` property behind the pref `layout.css.motion-path-offset-position.enabled`.
- To view the example containing the `ray()` function, the pref `layout.css.motion-path-ray.enabled` will also need to be enabled
- The code example is based on the (expected) behavior I tested in Nightly. I did not see the expected behavior of `normal` in Firefox beta 115.0b9.

### Related issues and pull requests

Doc tracker for this update: https://github.com/mdn/content/issues/27174
Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1559232

### To Dos

- [x] Update BCD: https://github.com/mdn/browser-compat-data/pull/20250
- [x] Update release notes: https://github.com/mdn/content/pull/27730
    - [x] Updated release note for Fx116: https://github.com/mdn/content/pull/28036
- [x] Update mdn/data: https://github.com/mdn/data/pull/673

dipikabh committed:
updates syntax and intro

dipikabh committed:
adds example

dipikabh committed:
fixes prettier errors

estelle commented on files/en-us/web/css/offset-position/index.md:
```diff
@@ -46,8 +47,10 @@ offset-position: unset;
 ### Values
 
+- `normal`
+  - : Indicates that the element does not have an offset starting position. This is the default value when the {{cssxref("path")}} CSS function is used to specify the value for the {{cssxref("offset-path")}} property. When the {{cssxref("ray")}} CSS function is used to specify the value for the `offset-path` property, the `normal` keyword defaults to `0px, 0px` of the containing block.
```
```suggestion
  - : Indicates that the element does not have an offset starting position. This is the default value when the {{cssxref("path")}} CSS function is used to specify the value for the {{cssxref("offset-path")}} property.
```

or possibly just:

```suggestion
  - : Indicates that the element does not have an offset starting position.
```

This is confusing for me. The values generally just list what the `normal` value does/means. So second suggestion.

The second sentence brings in `path()` and `offset-path`, but the reader doesn't know what any of these things mean, and the spec and page say the initial  or default value is `auto`. likely we should not include this second sentence on this page, but rather, explain it in the example, guide, or additional narrative.

The last sentence should be on the `ray()` page,. I we include it here, it should be in the example. (Also, would be `0 0` or "top left" as text, i think, if we did include it)

estelle commented on files/en-us/web/css/offset-position/index.md:
```diff
@@ -46,8 +47,10 @@ offset-position: unset;
+  - : Indicates that the element does not have an offset starting position. This is the default value when the {{cssxref("path")}} CSS function is used to specify the value for the {{cssxref("offset-path")}} property. When the {{cssxref("ray")}} CSS function is used to specify the value for the `offset-path` property, the `normal` keyword defaults to `0px, 0px` of the containing block.
 - `auto`
-  - : The initial position is the position of the box specified by the {{cssxref("position")}} property.
+  - : Indicates that the offset starting position is the top-left corner of the element's box. This is the default value when `ray()` is used to specify the value for the {{cssxref("offset-path")}} property.
```
```suggestion
  - : Indicates that the offset starting position is the top-left corner of the element's box.
```
This is always the default value, so second sentence isn't needed. Includig the ray() "exception" makes it sound like it's an exception. so best to not include this.

dipikabh replied:
You're right. I've just left the part to say that this is the default value.

estelle commented on files/en-us/web/css/offset-position/index.md:
````diff
@@ -61,10 +64,121 @@ offset-position: unset;
+This example visually compares the initial offset starting position of an element when `ray()` is used to specify a value for the {{cssxref("offset-path")}} property. The `+` text inside the element box is used to depict the element's anchor point.
+
+```html hidden
+<div id="wrap">
+  <div class="box">+</div>
+  <div class="box box0">+</div>
+</div>
+
+<div id="wrap">
+  <div class="box">+</div>
+  <div class="box box1">+</div>
+  <p>offset-position: normal;</p>
+</div>
+
+<div id="wrap">
+  <div class="box">+</div>
+  <div class="box box2">+</div>
+  <p>offset-position: auto;</p>
+</div>
+
+<div id="wrap">
+  <div class="box">+</div>
+  <div class="box box3">+</div>
+  <p>offset-position: left top;</p>
+</div>
````
```suggestion
<div class="wrap">
  <div class="box">+</div>
  <div class="box box0">+</div>
</div>

<div class="wrap">
  <div class="box">+</div>
  <div class="box box1">+</div>
  <p>offset-position: normal;</p>
</div>

<div class="wrap">
  <div class="box">+</div>
  <div class="box box2">+</div>
  <p>offset-position: auto;</p>
</div>

<div class="wrap">
  <div class="box">+</div>
  <div class="box box3">+</div>
  <p>offset-position: left top;</p>
</div>

<div class="wrap">
  <div class="box">+</div>
  <div class="box box4">+</div>
  <p>offset-position: 25% 75%;</p>
</div>
```

ids must be unique to a page

dipikabh replied:
Of course! I think I started building out the example with an id and switched to use class later and resulted with a copy paste error. Thanks for catching.

estelle commented on files/en-us/web/css/offset-position/index.md:
````diff
@@ -61,10 +64,121 @@ offset-position: unset;
+```
+
+```css hidden
+#wrap {
````
```suggestion
.wrap {
```

estelle commented on files/en-us/web/css/offset-position/index.md:
```diff
@@ -61,10 +64,121 @@ offset-position: unset;
+  offset-position: left top;
+  offset-path: ray(0deg);
+  opacity: 100%;
+}
```
```suggestion
}

.box4 {
  offset-position: 25% 75%;
  offset-path: ray(0deg);
  opacity: 100%;
}
```

estelle replied:
adding a non-default value. This example seemed like a ray() example, adding a different value makes it more relevant to offset-position

estelle commented on files/en-us/web/css/offset-position/index.md:
```diff
@@ -61,10 +64,121 @@ offset-position: unset;
-### Setting initial offset position
+### Comparing various offset starting positions
+
+This example visually compares the initial offset starting position of an element when `ray()` is used to specify a value for the {{cssxref("offset-path")}} property. The `+` text inside the element box is used to depict the element's anchor point.
```
the ray example should come second as ray() is not supported in browsers yet and the other example works in most browsers

estelle reviewed (changes requested):
i think some of the changes belong in the ray() page or the offset-path page, rather than here

dipikabh committed:
fixes review comments part 1

dipikabh committed:
update ray() page and offset-position description

dipikabh committed:
fixes review comments part 2

dipikabh commented:
Thanks, @estelle, for your valuable feedback. I have fixed all your comments. Additionally:

On the `offset-position` page:
- Added a bit more info to intro paragraphs.
- Updated the "Result" text for the example "Comparing various offset starting positions" to also explain the `offset-position: normal` value.

On the `ray()` page:
- Replaced the link to the spec in the intro para with the link to `offset-position` property page.
- Added a note about `offset-position: auto` and `offset-position: normal` values.

estelle commented on files/en-us/web/css/offset-position/index.md:
```diff
@@ -75,6 +96,9 @@ offset-position: unset;
   width: 40px;
   height: 40px;
   background: cyan;
+  position: absolute;
+  left: 20px;
+  top: 20px;
```
on lines 81 and 84, if you make it `0%, 10%` and `90%, 100%` respectively, it will take a "break" at the ends so users can see the effect (or lack thereof in my case) of off-set position.

I am still trying to figure out why it's not working for me....

dipikabh replied:
> That said, I can't get this property to work in the playground.
:) Glad I could show you a demo of the feature working in Fx116 in our 1:1

> if you make it `0%, 10%` and `90%, 100%` respectively
I updated this to `0%, 20%` and `80%, 100%` as you explained in the 1:1

dipikabh replied:
> You could change it to 20px 20px maybe?

So there is a problem with `offset-path: path()`; `path()` currently is not respecting any of the other values for `offset-position`. I tried changing values for `offset-position` but the motion is always starting from top left. I thought at least `normal` would work. In fact, there is no effect of even removing `offset-position` altogether. I am leaving the value `left top` for now (I know this is misleading but it best matches where the path is starting now). I'll raise it as a bug.

estelle commented on files/en-us/web/css/offset-position/index.md:
```diff
@@ -87,6 +111,117 @@ offset-position: unset;
+}
+
+.box0 {
+  opacity: 100%;
```
add either a offset path or an offset position to this example, to show the effect (or lack thereof) of one without the other

dipikabh replied:
Added `offset-position: normal;`

estelle commented on files/en-us/web/css/offset-position/index.md:
```diff
@@ -87,6 +111,117 @@ offset-position: unset;
+<div class="wrap">
+  <div class="box">+</div>
+  <div class="box box1">+</div>
+  <p>offset-position: normal;</p>
```
Either number the boxes or write out both CSS properties that impact the element

```
  <p>box1: offset-position: normal;</p>
```

or

```
  <p>offset-position: normal;<br/>offset-path: ray(0deg);</p>
```

dipikabh replied:
Numbered the boxes and also displaying `ray()` values now along with each box

estelle commented on files/en-us/web/css/offset-position/index.md:
```diff
@@ -87,6 +111,117 @@ offset-position: unset;
+
+.box4 {
+  offset-position: 30% 70%;
+  offset-path: ray(180deg);
```
They all look the same to me, except this one is rotated. I am thinking it might be good to give them all the same ray, of maybe 10deg, or if a ray change makes a huge difference, duplicate box4, with the same non-default OPos but very different Opath

dipikabh replied:
I kept the ray angle as `0deg` because it is the easiest to understand (it is for me :) ) without adding any more variables to see what `offset-position` is doing. I changed the angle for the last one to `120deg` so there is some notion of an angle instead of just a flip.

I've also now added `background-clip` and `dashed` border to fix accessibility issues you illustrated in our 1:1.

estelle commented on files/en-us/web/css/ray/index.md:
```diff
@@ -9,9 +9,11 @@ browser-compat: css.types.ray
 {{CSSRef}}{{SeeCompatTable}}
 
-The **`ray()`** [CSS](/en-US/docs/Web/CSS) [function](/en-US/docs/Web/CSS/CSS_Functions) defines a line segment that begins from an [offset starting position](https://drafts.fxtf.org/motion/#offset-starting-position) and extends in the direction of the specified angle. The line segment is referred to as "ray". The length of a ray can be constrained by specifying a size and using the `contain` key ...
+The **`ray()`** [CSS](/en-US/docs/Web/CSS) [function](/en-US/docs/Web/CSS/CSS_Functions) defines a line segment that begins from an {{cssxref("offset-position")}} and extends in the direction of the specified angle. The line segment is referred to as "ray". The length of a ray can be constrained by specifying a size and using the `contain` keyword.
```
move the css example, that comes right after the syntax in ##syntax between the first and second paragraphs.

estelle reviewed (commented):
i can't get the offset-position to work, so kind of blindly reviewing...

dipikabh commented on files/en-us/web/css/offset-position/index.md:
```diff
@@ -87,6 +111,117 @@ offset-position: unset;
+  align-items: center;
+  width: 50px;
+  height: 50px;
+}
```
Done. Thanks for the tip 👍

dipikabh committed:
fixes review comments part 3

dipikabh commented:
Thanks, @estelle, for all your hints and suggestions. It has helped improve the examples more. I have addressed all your comments.

Btw, this is what I get now in the "Result" frame (the right screenshot is for when Achromatopsia (no color) is selected in the Accessibility inspector in dev tools):

<div style="display: flex;">
  <img src="https://github.com/mdn/content/assets/23019147/8ed8511d-8931-4854-a5d6-ffb86a75c1dc" alt="fx116-offset-position" width="40%")>
  <img src="https://github.com/mdn/content/assets/23019147/ea4952a0-67fd-440c-9ad6-e8190ad529ec" alt="fx116-offset-position-achromatopsia" width="40%")>
</div>

estelle commented on files/en-us/web/css/ray/index.md:
```diff
@@ -9,17 +9,19 @@ browser-compat: css.types.ray
 
+The `ray()` function is used in [CSS motion path](/en-US/docs/Web/CSS/CSS_motion_path). It is used as a value for the [`offset-path`](/en-US/docs/Web/CSS/offset-path) property to define the path that an animated element can follow. The element is initially positioned by moving the element's [`offset-anchor`](/en-US/docs/Web/CSS/offset-anchor) point to the path's offset starting position.
+
+> **Note:** The default offset starting position of a ray is the `left top` corner (or `0 0`) of the element box (that is, `offset-position` is `auto`). With `offset-position: normal`, the starting position of the ray is `50%, 50%` of the containing block.
```
```suggestion
> **Note:** The default offset starting position of a ray is `auto`. If `offset-position: auto` is explicitly or allowed to default, the offset starting position is the `left top` corner (or `0 0`) of the element box. With `offset-position: normal`, the starting position of the ray is `50%, 50%` of the containing block.
```

Wordier, but I find this easier to understand.

dipikabh replied:
I just checked today - there is another Fx issue in this release for which this note will need to be removed - a new parameter getting added to ray()'s syntax. I'll leave this as is for now.

dipikabh replied:
I went ahead and incorporated the suggested change after all. As long as this is out there before I come out with an updated version of the page, the text should be clear. Thanks 👍

estelle commented on files/en-us/web/css/offset-position/index.md:
```diff
@@ -78,15 +99,142 @@ offset-position: unset;
+
+### Comparing various offset starting positions
+
+This example visually compares the initial offset starting position of an element when `ray()` is used to specify a value for the {{cssxref("offset-path")}} property. The `+` text inside the element box is used to depict the element's anchor point.
```
```suggestion
This example visually compares the initial offset starting position of an element when `ray()` is used to specify a value for the {{cssxref("offset-path")}} property. The number included inside the element box is used both to indicate which CSS block applied to which box and to depict the element's anchor point.
```

dipikabh replied:
Thanks for catching this change 👍. I've update it to:

> The number inside the element box indicates the element to which CSS is applied as well as the element's anchor point.

estelle reviewed (approved):
Approving. Just very minor suggested edits.

dipikabh commented on files/en-us/web/css/offset-position/index.md:
```diff
@@ -46,8 +49,10 @@ offset-position: unset;
-  - : The initial position is the position of the box specified by the {{cssxref("position")}} property.
+  - : Indicates that the offset starting position is the top-left corner of the element's box. This is the default value.
 - `<position>`
   - : A {{cssxref("&lt;position&gt;")}}. A position defines an x/y coordinate, to place an item relative to the edges of an element's box. It can be defined using one to four values. If two non-keyword values are used, the first value represents the horizontal position and the second represents the vertical position. If only one value is specified, the second value is assumed to be `center`. If t ...
```
Done

dipikabh committed:
fixes review comments part 4
